packages/headless/src/api/analytics/search-analytics.test.ts

```diff
@@ -471,5 +471,114 @@ describe('#configureLegacyAnalytics', () => {
         ...metadata,
       });
     });
+
+    describe('getBaseMetadata with Generative Answering ID', () => {
+      beforeEach(() => {
+        vi.clearAllMocks();
+      });
+
+      it('should include generativeQuestionAnsweringId when available from answer service', () => {
+        const mockState = {
+          ...getBaseState(),
+          generatedAnswer: {
+            ...getGeneratedAnswerInitialState(),
+            answerConfigurationId: 'test-config-id',
+          },
+        };
+
+        const provider = new SearchAnalyticsProvider(() => mockState);
+        const metadata = provider.getBaseMetadata();
+
+        expect(metadata).toBeDefined();
+        expect(metadata).toHaveProperty('coveoHeadlessVersion');
+      });
+
+      it('should include generativeQuestionAnsweringId from search response when the answerId is unavailable', () => {
+        const stateWithSearchId = {
+          ...getBaseState(),
+          generatedAnswer: {
+            ...getGeneratedAnswerInitialState(),
+            answerConfigurationId: undefined, // No answer config
+          },
+          search: {
+            ...buildMockSearchState({}),
+            response: {
+              ...buildMockSearchState({}).response,
+              extendedResults: {
+                generativeQuestionAnsweringId: 'search-response-id-456',
+              },
+            },
+          },
+        };
+
+        const provider = new SearchAnalyticsProvider(() => stateWithSearchId);
+        const metadata = provider.getBaseMetadata();
+
+        expect(metadata).toBeDefined();
+        expect(metadata).toHaveProperty('coveoHeadlessVersion');
+      });
+
+      it('should not include generativeQuestionAnsweringId when not available', () => {
+        const stateWithoutIds = {
+          ...getBaseState(),
+          generatedAnswer: {
+            ...getGeneratedAnswerInitialState(),
+            answerConfigurationId: undefined,
+          },
+          search: {
+            ...buildMockSearchState({}),
+            response: {
+              ...buildMockSearchState({}).response,
+              extendedResults: {
+                // N/A
+              },
+            },
+          },
+        };
+
+        const provider = new SearchAnalyticsProvider(() => stateWithoutIds);
+        const metadata = provider.getBaseMetadata();
+
+        expect(metadata).toBeDefined();
+        expect(metadata).toHaveProperty('coveoHeadlessVersion');
+        expect(metadata).not.toHaveProperty('generativeQuestionAnsweringId');
+      });
+
+      it('should handle partial search states correctly', () => {
```

Comment:
Can you find a way to be a bit more descriptive with the title of the test here?
Not sure what is the expectation from just reading the test name `handle...correctly` doesn't really tell me what is supposed to happen.